swift/storage.py

```diff
@@ -84,15 +83,17 @@ def validate_settings(backend):
             raise ImproperlyConfigured("SWIFT_PROJECT_ID or SWIFT_PROJECT_NAME must \
              be defined when using version 3 auth")
 
-    # Encode temp_url_key as a latin-1 string (hmac does not accept unicode keys)
+    # Validate temp_url parameters
     if backend.use_temp_urls:
         if backend.temp_url_key is None:
             raise ImproperlyConfigured("SWIFT_TEMP_URL_KEY must be set when \
              SWIFT_USE_TEMP_URL is True")
+
+        # Encode temp_url_key as bytes
         try:
-            backend.temp_url_key = backend.temp_url_key.encode('latin-1')
+            backend.temp_url_key = backend.temp_url_key.encode()
```

Comment:
`encode('ascii')` so python 3 don't use utf-8. This will of course assume that we always configure the key as a string. I think that's acceptable.